src/realm/obj.cpp

```diff
@@ -977,6 +981,12 @@ FullPath Obj::get_path() const
     return result;
 }
 
+void Obj::add_index(Path& path, Index index) const noexcept
```

Comment:
same here. `std::vector<T>::emplace_back()` can throw and also `std::string` constructor.